lib/FusionInventory/Agent/Task/Inventory/Win32/Softwares.pm

```diff
@@ -316,4 +339,55 @@ sub _processMSIE {
 
 }
 
+# List of SQL Instances
+sub _getSqlInstancesList {
+    my (%params) = @_;
+
+    my $softwareName = $params{softwarename};
+    my $softwareVersion = $params{softwareversion};
+
+    # Registry access for SQL Instances
+    my $sqlinstancesList = getRegistryKey(
+        path    => "HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Microsoft SQL Server/Instance Names/SQL",
+        %params
+    ) or return;
+
+    # List of SQL Instances
+    foreach my $sqlinstance (keys %$sqlinstancesList) {
+        my $sqlinstanceName = $sqlinstance;
+        my $sqlinstanceValue = $sqlinstancesList->{$sqlinstance};
+        # Get version and edition for each instance
+        my ($sqlinstanceEditionValue,$sqlinstanceVersionValue) = _getSqlInstancesVersions(
+            %params,
+            SOFTVERSION => $softwareVersion,
+            NAME        => $sqlinstanceName,
+            VALUE       => $sqlinstanceValue
+        );
+        next unless $sqlinstanceEditionValue && $sqlinstanceVersionValue;
+        return $sqlinstanceEditionValue
+            if ($softwareVersion eq $sqlinstanceVersionValue); 
+    }
+}
+
+# SQL Instances versions
+# Return version and edition for each instance
+sub _getSqlInstancesVersions {
+    my (%params) = @_;
+
+    my $softwareVersion = $params{SOFTVERSION};	
+    my $sqlinstanceName = $params{NAME};
+    my $sqlinstanceValue = $params{VALUE};
+
+    my $sqlinstanceVersions = getRegistryKey(
+        path    => "HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Microsoft SQL Server/" . $sqlinstanceValue . "/Setup",
+        %params
+    );
+
+    my $sqlinstanceVersionValue = $sqlinstanceVersions->{'/Version'} or return;
+    my $sqlinstanceEditionValue = $sqlinstanceVersions->{'/Edition'};
+    # If software version match instance one
+    return ($sqlinstanceEditionValue,$sqlinstanceVersionValue)
+        if ($softwareVersion eq $sqlinstanceVersionValue);
+}
```

Comment:
same comment as for line 370: no explicit default return value.

Reply:
@keyser75000 
Try replacing line 389 & 390 by:
```
    return if ($softwareVersion ne $sqlinstanceVersionValue);
    return ($sqlinstanceEditionValue,$sqlinstanceVersionValue);
```

Reply:
Finally, if only $sqlinstanceEditionValue needs to be returned, you can replace lines 386-390 by:
```
    return unless ($sqlinstanceVersions->{'/Version'} && $sqlinstanceVersions->{'/Version'} eq $softwareVersion);
    return $sqlinstanceVersions->{'/Edition'};
```